### DISCUSSION TITLE
Can generator be used in an offline environment?

### DISCUSSION THREAD

**Discussion by canadabri** [prometheus/snmp_exporter#1276] [Q&A] [answered]
I am quite inexperienced with this connector.  I am wondering if the generator function can be tweaked to work in an offline environment.  From what I can tell in the makefile, there are a number of default/standard MIBs that get downloaded/updated and placed in the mibs folder.

Is there a way that I can just manually copy any and all MIB files I need to the mibs folder, edit the generator.xml file with my auth details, and build the snmp.yml file for my offline environment?

I have tried to do this, but the first command `make generator mibs` fails - and I am assuming it is because the process assumes that the environment has an internet connection.   Has anyone done this and can provide info on how to do it?

**Comment by canadabri** [comment]
If offline is not possible, perhaps I could do the generate function from a computer connected to the internet, then copy my snmp.yml file to my offline environment?

If that is possible, I am still running into errors with the first step:  `make generator mibs`

This is returning an error: 
```sh
myuser@host ~/git/snmp_exporter/generator $ make generator mibs
go build
go: errors parsing go.mod:
/home/myuser/git/snmp_exporter/go.mod:5: unknown directive: toolchain
make: *** [Makefile:90: generator] Error 1
```

**Reply by SuperQ** [reply]
Your Go compiler is too old.

**Reply by canadabri** [reply]
Thank you.   I was able to get my go compiler updated, and that did let me run the make command.

**Comment by SuperQ** [comment, marked as answer]
Yes, you can download all the MIBs and copy them to the offline environment. 

The `make mibs` is there because we want to make sure our system provides reproducible builds. But we technically can't include the MIB files in our repo because most MIB files are copyrighted and don't have an open-source compatible redistribution license.

**Reply by canadabri** [reply]
Thank you.   I am using my internet connected computer as my staging area, and building my yml file on it, then copying the resulting file to my offline environment.

This has gotten me further towards a working setup, but I am running into a new problem that should probably be a new discussion thread.
